src/main/java/com/minecolonies/coremod/entity/ai/mobs/AbstractArcherBarbarian.java

```diff
@@ -0,0 +1,302 @@
+    /* default */ private static final DataParameter<Boolean>           SWINGING_ARMS         =
+      EntityDataManager.<Boolean>createKey(AbstractArcherBarbarian.class, DataSerializers.BOOLEAN);
+    /* default */ private final        EntityAIAttackRangedBowBarbarian aiArrowAttack         = new EntityAIAttackRangedBowBarbarian(this, 1.0D, 20, 15.0F);
+    /* default */ private static final float                            ENTITY_WIDTH          = 0.6F;
+    /* default */ private static final float                            ENTITY_HEIGHT         = 1.99F;
+    /* default */ private static final int                              BARBARIAN_BASE_HEALTH = 25;
+    /* default */ private static final int                              DIFFICULTY_SUBTRACTER = 14;
+    /* default */ private static final int                              DIFFICULTY_MULTIPLIER = 4;
+    /* default */ private static final int                              BOW_DAMAGE            = 20;
+    /* default */ private static final int                              BOW_DAMAGE_HARD       = 40;
+    /* default */ private static final double                           GRAVITY_MULTIPLIER    = 0.20000000298023224D;
+    /* default */ private static final float                            ARROW_VELOCITY        = 1.6F;
+    /* default */ private static final float                            SOUND_MULTIPLIER      = 0.4F;
+    /* default */ private static final float                            SOUND_ADDER           = 0.8F;
+    /* default */ private static final float                            ENTITY_EYE_HEIGHT     = 1.74F;
+
+    private final EntityAIAttackMelee aiAttackOnCollide = new EntityAIAttackMelee(this, 1.2D, false)
+    {
+        /**
+         * Resets the task
+         */
+        public void resetTask()
+        {
+            super.resetTask();
+            AbstractArcherBarbarian.this.setSwingingArms(false);
+        }
+
+        /**
+         * Execute a one shot task or start executing a continuous task
+         */
+        public void startExecuting()
+        {
+            super.startExecuting();
+            AbstractArcherBarbarian.this.setSwingingArms(true);
+        }
+    };
+
+    public AbstractArcherBarbarian(final World worldIn)
+    {
+        super(worldIn);
+        this.setSize(ENTITY_WIDTH, ENTITY_HEIGHT);
+        this.setCombatTask();
+    }
+
+    protected void initEntityAI()
+    {
+        this.tasks.addTask(1, new EntityAISwimming(this));
+        this.tasks.addTask(2, new EntityAIRestrictSun(this));
+        this.tasks.addTask(3, new EntityAIAvoidEntity(this, EntityWolf.class, 6.0F, 1.0D, 1.2D));
+        this.tasks.addTask(5, new EntityAIWalkToRandomHuts(this, 2.0D));
+        this.tasks.addTask(6, new EntityAIWatchClosest(this, EntityPlayer.class, 8.0F));
+        this.tasks.addTask(6, new EntityAILookIdle(this));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 6 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)